modules/nw-openstack-external-ccm.adoc

```diff
@@ -0,0 +1,28 @@
+// Module included in the following assemblies:
+//
+// tbd
+// TODO: GitHub link clearance.
+
+:_content-type: CONCEPT
+[id="nw-openstack-external-ccm_{context}"]
+= The {rh-openstack} Cloud Controller Manager (CCM)
+
+In {product-title} 4.12, clusters that run on {rh-openstack-first} are switched from the legacy {rh-openstack} cloud provider to the external link:https://github.com/kubernetes/cloud-provider-openstack[OpenStack CCM]. This change follows the move in Kubernetes from in-tree, legacy cloud providers to external cloud providers that are implemented by using the link:https://kubernetes.io/docs/concepts/architecture/cloud-controller/[Cloud Controller Manager].
+
+To preserve user-defined configurations for the legacy cloud provider, existing configurations are mapped to new ones as part of the migration process. It searches for configurations in the following places:
+
+* `kube-cloud-config` in the `openshift-config-managed` namespace
+
+* `cloud-provider-config` in the `openshift-config` namespace
+
+Configurations that are found are synchronized to the `cloud-conf` config map in the `openshift-cloud-controller-manager` namespace.
+
+As part of this synchronization, the OpenStack CCM Operator alters the new config map such that its properties are compatible with the external cloud provider. The file is changed in the following ways:
+
+* The `[Global] secret-name`, `[Global] secret-namespace`, and `[Global] kubeconfig-path` options are removed. They do not apply to the external cloud provider.
+
+* The `[Global] use-clouds`, `[Global] clouds-file`, and `[Global] cloud` options are added.
+
+* The entire `[BlockStorage]` section is removed. External cloud providers no longer perform storage operations.
+
+* The `[LoadBalancer] use-octavia` and `[LoadBalancer] enabled` options are set or added according to the wider cluster configuration.
```

Comment:
use-octavia is always set to True, perhaps there is no need to mention it?

Reply:
```suggestion
* The `[LoadBalancer] enabled` option is set or added according to the wider cluster configuration.
```